### PR TITLE
fixed errors and structure

### PR DESCRIPTION
**Pull request by malminawi**
Please review the code

### CODE REVIEW

File: authBackEnd/apps/common/mixins.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
 from django.shortcuts import redirect, reverse
 
 
-class LoginRequiredMixin(object):  # these are kinda mysterious - how the fuck do they work / what do they do?
```
Comment (malminawi) [inline authBackEnd/apps/common/mixins.py:4]: The login MIxin is basically to make sure that the user is signed in before viewing the dashboard. It won't let the user to see the dashboard without being logged in

Comment (NullDefault) [reply authBackEnd/apps/common/mixins.py:4]: Makes sense!

File: authBackEnd/apps/common/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+from django.utils.translation import gettext as _
+
+
+class TimeStamp(models.Model):
```
Comment (NullDefault) [inline authBackEnd/apps/common/models.py:5]: This should probably be moved to the api models and all other models should subclass from it.

File: authBackEnd/apps/common/views.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.shortcuts import render
```
Comment (NullDefault) [inline authBackEnd/apps/common/views.py]: There's a bunch of empty files -> would it be an issue if they were deleted?

File: authBackEnd/apps/dashboard/admin.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (NullDefault) [inline authBackEnd/apps/dashboard/admin.py]: same thing as above

File: authBackEnd/apps/dashboard/apps.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class DashboardConfig(AppConfig):
+    name = 'apps.dashboard'
```
Comment (NullDefault) [inline authBackEnd/apps/dashboard/apps.py:5]: can you rename it to something other than apps.dashboard? I feel like something like auth.dashboard would work better

File: authBackEnd/apps/dashboard/urls.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path, include
+from .views import DashboardView
+
+urlpatterns = [
+    path('', DashboardView.as_view(), name='dashboard')
+]
+app_name = 'dashboard'
```
Comment (NullDefault) [inline authBackEnd/apps/dashboard/urls.py]: what is this used for

File: authBackEnd/apps/dashboard/urls2.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path, include, re_path
+from .views import DashboardView, AppView
+
+urlpatterns = [
+    re_path(r'^.*$', AppView.as_view(), name='app'),
+]
```
Comment (NullDefault) [inline authBackEnd/apps/dashboard/urls2.py]: why are there two url files? they should be combined

File: authBackEnd/apps/sadmin/apps.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class AdminConfig(AppConfig):
+    name = 'apps.sadmin'
```
Comment (NullDefault) [inline authBackEnd/apps/sadmin/apps.py]: auth.super_admin perhaps?

File: authBackEnd/apps/user/admin.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from authBackEnd.apps.user import models as account_models
+
```
Comment (NullDefault) [inline authBackEnd/apps/user/admin.py:3]: use an array and a for loop -> it's cleaner and scales better

File: authBackEnd/apps/user/forms.py [hunk 10]
```diff
@@ -1,28 +1,24 @@
...
+        fields = ('first_name', 'last_name', 'school_id', 'graduation_class', 'role')
+        required = ('first_name', 'last_name', 'school_id', 'graduation_class', 'role')
 
     def clean(self):
         cleaned_data = super().clean()
         referral = cleaned_data['referral_code']
         role = cleaned_data['role']
         if not self.validate_referral(referral, role):
-            self.add_error('referral_code', 'Referral code is not valid or is already used!')
+            self.add_error('referral_code', 'Referal code is not valid or is already used!')
```
Comment (NullDefault) [inline authBackEnd/apps/user/forms.py]: typo on Referal

File: authBackEnd/apps/user/models.py [hunk 11]
```diff
@@ -46,15 +38,16 @@ def create_superuser(self, email, password, **extra_fields):
         return self._create_user(email, password, **extra_fields)
 
 
-class User(auth_models.AbstractBaseUser, auth_models.PermissionsMixin, TimeStamp):
-    # TODO: This needs to be integrated with the api user model
-    """ User Model for authentication and manage account """
+class User(auth_models.AbstractBaseUser, auth_models.PermissionsMixin, common_models.TimeStamp):
```
Comment (NullDefault) [inline authBackEnd/apps/user/models.py:41]: We need to figure out how this connects to the db models

File: authBackEnd/apps/user/models.py [hunk 12]
```diff
@@ -65,16 +58,16 @@ class User(auth_models.AbstractBaseUser, auth_models.PermissionsMixin, TimeStamp
...
     def __str__(self):
         return f'{self.first_name} {self.last_name} - {self.email}'
 
 
-class Referral(TimeStamp):
-    """ Referral model to store referrals """
-    code = models.CharField('Referral Code', max_length=255, unique=True, default=generate_referral_code)
-    role = models.SmallIntegerField('Role', choices=ROLE_CHOICES, null=True)
+class Referral(common_models.TimeStamp):
+    """ Refferal model to store referals """
```
Comment (NullDefault) [inline authBackEnd/apps/user/models.py]: typo

File: authBackEnd/templates/base.html [hunk 13]
```diff
@@ -0,0 +1,52 @@
+{% load static %}
```
Comment (NullDefault) [inline authBackEnd/templates/base.html:1]: I can't read through all the html code so I trust you know what it does.

File: config/urls.py [hunk 14]
```diff
@@ -20,8 +20,8 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('accounts/', include('allauth.urls')),
-
-    path('', include('api.settings.urls')),
-    path('', include('frontend.urls')),
```
Comment (NullDefault) [inline config/urls.py:25]: why did you remove the frontend and api urls?

File: frontend/.gitignore [hunk 15]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
```
Comment (NullDefault) [inline frontend/.gitignore]: there is no need for a second .gitignore
why is it here?

File: frontend/README.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (NullDefault) [inline frontend/README.md]: why is this here?

File: frontend/package.json [hunk 17]
```diff
@@ -1,37 +1,42 @@
 {
-  "name": "frontend",
-  "version": "1.0.0",
-  "description": "",
-  "main": "App.js",
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "dev": "webpack --mode development src/App.js --output ./static/build/main.js",
```
Comment (NullDefault) [inline frontend/package.json:8]: I don't understand what is happening here. Why are you overriding the frontend code?

File: frontend/public/index.html [hunk 18]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (NullDefault) [inline frontend/public/index.html:1]: what is this? does the old frontend work?

File: frontend/public/manifest.json [hunk 19]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
```
Comment (NullDefault) [inline frontend/public/manifest.json]: if you're gonna put this here please override the default values

File: frontend/src/components/elements/NavBar.js [hunk 20]
```diff
@@ -2,35 +2,37 @@ import Navbar from "react-bootstrap/Navbar";
 import {Nav, NavItem} from "react-bootstrap";
 import React from "react";
 import Link from "react-router-dom/Link";
-import "../../../static/css/components/navbar.css";
-
+import "../../static/css/components/navbar.css";
+import img from "../../static/assets/volog_logo/green_on_gray.png"
```
Comment (NullDefault) [inline frontend/src/components/elements/NavBar.js]: img is not a descriptive variable name

File: frontend/src/index.js [hunk 21]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (NullDefault) [inline frontend/src/index.js:1]: is this what renders the frontend now?

File: frontend/src/pages/MentorDashboard.js [hunk 22]
```diff
@@ -1,11 +1,40 @@
...
-            MENTOR DASHBOARD
-        </div>
+class MentorDashboard extends Component {
+     constructor(props) {
+    super(props);
+
+    this.state = {isLoading: true};
+  }
+    componentDidMount() {
+        let com = this
```
Comment (NullDefault) [inline frontend/src/pages/MentorDashboard.js:12]: comments would be helpful here

**Review by NullDefault — CHANGES_REQUESTED** [review]
comments are attached

File: authBackEnd/apps/common/models.py (hunk 2 above)
Comment (malminawi) [reply authBackEnd/apps/common/models.py:5]: The TimeStamp model can be imported and inherited/used in the api models... Done

File: authBackEnd/apps/common/views.py (hunk 3 above)
Comment (malminawi) [reply authBackEnd/apps/common/views.py]: Completed

File: authBackEnd/apps/dashboard/admin.py (hunk 4 above)
Comment (malminawi) [reply authBackEnd/apps/dashboard/admin.py]: Completed

File: authBackEnd/apps/dashboard/apps.py (hunk 5 above)
Comment (malminawi) [reply authBackEnd/apps/dashboard/apps.py:5]: the apps in the apps.dashboard is the apps folder inside the authBackEnd folder, I think it will not make any difference to change the name.

File: authBackEnd/apps/dashboard/urls.py (hunk 6 above)
Comment (malminawi) [reply authBackEnd/apps/dashboard/urls.py]: This is the first page that the user will be directed after signing in. in this case, it will be changed to the frontend react app. (still in progress). this dashboard will be deleted, as I created this before the frontend was created to test what I was doing.

File: authBackEnd/apps/dashboard/urls2.py (hunk 7 above)
Comment (malminawi) [reply authBackEnd/apps/dashboard/urls2.py]: completed

File: authBackEnd/apps/user/admin.py (hunk 9 above)
Comment (malminawi) [reply authBackEnd/apps/user/admin.py:3]: What do you mean? where?

File: authBackEnd/apps/sadmin/apps.py (hunk 8 above)
Comment (malminawi) [reply authBackEnd/apps/sadmin/apps.py]: it was renamed to superAdmin

File: authBackEnd/apps/user/forms.py (hunk 10 above)
Comment (malminawi) [reply authBackEnd/apps/user/forms.py]: completed

File: authBackEnd/apps/user/models.py (hunk 12 above)
Comment (malminawi) [reply authBackEnd/apps/user/models.py]: completed

File: frontend/.gitignore (hunk 15 above)
Comment (malminawi) [reply frontend/.gitignore]: completed

File: frontend/README.md (hunk 16 above)
Comment (malminawi) [reply frontend/README.md]: deleted

File: frontend/package.json (hunk 17 above)
Comment (malminawi) [reply frontend/package.json:8]: when we use the create react app, it auto generate this file...

File: frontend/public/index.html (hunk 18 above)
Comment (malminawi) [reply frontend/public/index.html:1]: this is a part of the create-react app

File: frontend/public/manifest.json (hunk 19 above)
Comment (malminawi) [reply frontend/public/manifest.json]: completed

File: frontend/src/components/elements/NavBar.js (hunk 20 above)
Comment (malminawi) [reply frontend/src/components/elements/NavBar.js]: completed

File: frontend/src/index.js (hunk 21 above)
Comment (malminawi) [reply frontend/src/index.js:1]: import React from 'react'; is from react app

File: frontend/src/pages/MentorDashboard.js (hunk 22 above)
Comment (malminawi) [reply frontend/src/pages/MentorDashboard.js:12]: This is for the user role based. It hits the api, if the user role is not valid, it will not get them to dashboard.
ex, if mentor is the user, it wont let them access the admin page. this applies for the student as well.
